modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/Fragment.java

```diff
@@ -142,8 +143,12 @@ private FragmentMapping mapping(MappingQueryContext ctx, RelMetadataQuery mq, Su
         try {
             FragmentMapping mapping = IgniteMdFragmentMapping._fragmentMapping(root, mq, ctx);
 
-            if (rootFragment())
-                mapping = FragmentMapping.create(ctx.localNodeId()).colocate(mapping);
+            if (rootFragment()) {
+                if (ctx.isLocal() && !(root instanceof IgniteTableModify))
```

Comment:
Why IgniteTableModify is excluded? ColocationGroup affects only scans as far as I know. We can't say to IgniteTableModify to filter local data, but it's absolutely fine to say it to IgniteTableModify's underlying scans. For example `DELETE FROM tbl WHERE a = 1` with local flag should only deal with data located on current node (but should delete backups on other nodes as well), and for example `INSERT INTO tbl1 ... AS SELECT FROM tbl2` with local flag should insert data to tbl1 to any node but only selected local data from tbl2.